engine/src/main/java/org/terasology/rendering/opengl/FrameBuffersManager.java

```diff
@@ -180,43 +264,50 @@ public void createOrUpdateFullscreenFbos() {
         new FBObuilder("sceneBlur0",      halfScale,    FBO.Type.DEFAULT).build();
         new FBObuilder("sceneBlur1",      halfScale,    FBO.Type.DEFAULT).build();
 
-        // buffers for the Final Post-Processing
         new FBObuilder("ocUndistorted",   fullScale,    FBO.Type.DEFAULT).build();
         new FBObuilder("sceneFinal",      fullScale,    FBO.Type.DEFAULT).build();
-
-        graphicState.refreshDynamicFBOs();
-        postProcessor.refreshDynamicFBOs();
     }
 
-    public void deleteFBO(String title) {
-        if (fboLookup.containsKey(title)) {
-            FBO fbo = fboLookup.get(title);
+    private void recreateShadowMapFBO() {
+        int shadowMapResFromSettings = renderingConfig.getShadowMapResolution();
+        Dimensions shadowMapResolution =  new Dimensions(shadowMapResFromSettings, shadowMapResFromSettings);
+        sceneShadowMap = new FBObuilder("sceneShadowMap", shadowMapResolution, FBO.Type.NO_COLOR).useDepthBuffer().build();
+        handleDisposedShadowMap(sceneShadowMap);
+    }
 
-            glDeleteFramebuffersEXT(fbo.fboId);
-            glDeleteRenderbuffersEXT(fbo.depthStencilRboId);
-            GL11.glDeleteTextures(fbo.normalsBufferTextureId);
-            GL11.glDeleteTextures(fbo.depthStencilTextureId);
-            GL11.glDeleteTextures(fbo.colorBufferTextureId);
+    private void handleDisposedShadowMap(FBO fbo) {
+        if (fbo.getStatus() == FBO.Status.DISPOSED) {
+            logger.warn("Failed to generate ShadowMap FBO. Turning off shadows.");
+            renderingConfig.setDynamicShadows(false);
         }
     }
 
+    // TODO: (?) move to the post-processor?
```

Comment:
Most of the screenshot-related methods could perhaps be moved into the post-processor, leaving in this manager only a method that returns the data held in the sceneFinal FBO as a ByteBuffer.